exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -492,10 +537,18 @@ private void writeDataAllText(MapWriter map, FieldSelection selection,
       }
     }
     map.end();
-
   }
 
   /**
+   * Puts copy of field path list to fieldPathWriter map.
+   * @param fieldName
+   */
+  private void putFieldPath(String fieldName, MapWriter map) {
+    List<String> fieldPath = Lists.newArrayList(path);
```

Comment:
Before allocating the fieldPath list, should we first check if it is already present in the fieldPathWriter ? This function is called for every null value which makes it quite expensive (as shown by your performance experiments).  If there are 1000 records of type  {'x': null}, ideally we want to call this method only for field 'x' once.